lib/puppet/indirector/catalog/static_compiler.rb

```diff
@@ -205,7 +206,7 @@ def store_content(resource)
       Puppet.info "Content for '#{resource[:source]}' already exists"
     else
       Puppet.info "Storing content for source '#{resource[:source]}'"
-      content = Puppet::FileServing::Content.indirection.find(resource[:source])
+      content = Puppet::FileServing::Content.indirection.find(resource[:source], {:environment => request.environment})
```

Comment:
The bug we ran into involved requesting a new file from an environment other than production, results in the puppet run failing because the requested file does not exist in production.

Per my original comment, I didn't add tests because the store_content method appears to just be mocked and can't say I'm familiar enough with rspec or puppet code to test it properly.